.circleci/config.yml

```diff
@@ -0,0 +1,100 @@
+version: 2
+jobs:
+  relay:
+    working_directory: ~/artsy/reaction
+    docker:
+    - image: circleci/node:8
+    steps: 
+      - checkout
+      - restore_cache:
+          keys:
+          - yarn-deps-{{ checksum "yarn.lock" }}
+          - yarn-deps-
+      - run: yarn --check-files
+      - run: yarn relay
+  lint:
+    working_directory: ~/artsy/reaction
+    docker:
+    - image: circleci/node:8
+    steps: 
+      - checkout
+      - restore_cache:
+          keys:
+          - yarn-deps-{{ checksum "yarn.lock" }}
+          - yarn-deps-
+      - run: yarn --check-files
+      - run: yarn lint 
+  type-check:
+    working_directory: ~/artsy/reaction
+    docker:
+    - image: circleci/node:8
+    steps: 
+      - checkout
+      - restore_cache:
+          keys:
+          - yarn-deps-{{ checksum "yarn.lock" }}
+          - yarn-deps-
+      - run: yarn --check-files
+      - run: yarn type-check
+  deploy:
+    working_directory: ~/artsy/reaction
+    docker:
+    - image: circleci/node:8
+    steps:
+      - checkout
+      - run: echo "//registry.npmjs.org/:_authToken=$NPM_TOKEN" >> ~/.npmrc
+      - restore_cache:
+          keys:
+          - yarn-deps-{{ checksum "yarn.lock" }}
+          - yarn-deps-
+      - run: yarn --check-files
+      - run: yarn semantic-release
+      - run: cp .env.oss .env
+      - run: yarn deploy-storybook
+  test:
+    working_directory: ~/artsy/reaction
+    docker:
+    - image: circleci/node:8
+    steps: 
+      - checkout
+      - restore_cache:
+          keys:
+          - yarn-deps-{{ checksum "yarn.lock" }}
+          - yarn-deps-
+      - run: yarn --check-files
+      - run: yarn test -w 4
```

Comment:
We were running jest tests inband (serially) before. The tests were pretty slow so I tried removing that restriction. Turns out, they eat a lot of memory and when too many run in parallel it starves out the CI. Restricting it to 4 seems to be a good balance of speed and conservative resource consumption.

Reply:
On a separate note, `yarn test -w 4` seems to blindly append the argument `-w 4` to the actual command in npm script, which is currently `node verify-node-version.js && jest`. _(I was surprised for a few seconds by how smart yarn is to apply the argument to the right command!)_ I wonder how we can avoid future mistakes when `jest`is no longer the last command.